### PR TITLE
Handle empty objects in `EqualKeys`

### PR DESCRIPTION
**Pull request by ruimarinho**
Closes #34.

### CODE REVIEW

**Comment by ruimarinho** [comment]
ping @nunofgs

File: test/asserts/equal-keys-assert_test.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ describe('EqualKeysAssert', () => {
     });
   });
 
-  it('should throw an error if an object does not have the expected keys', () => {
+  it('should throw an error if the object not have the expected keys', () => {
```
Comment (nunofgs) [inline test/asserts/equal-keys-assert_test.js]: `does not`?

**Comment by nunofgs** [comment]
Hey guys, not trying to be difficult but I'm still having trouble with this concept. I personally dislike mixing "empty object" validation with key matching but I will of course capitulate if the intent is to follow the _should.js_ implementation (as mentioned [here](https://github.com/seegno/validator.js-asserts/pull/34#issuecomment-168539426)).

Having said that, I believe the following should **not** throw an error:

```js
// This throws an error.
new Assert().EqualKeys().validate({});

// Since this also does not throw an error.
({}).should.have.keys();
```

If you fix that issue, I have no further objections.

/cc @ricardogama, @ruimarinho

**Comment by nunofgs** [comment]
Also, regardless of the decision, should we also update the README with a description of the breaking change?

File: test/asserts/equal-keys-assert_test.js [hunk 2]
```diff
@@ -45,9 +45,49 @@ describe('EqualKeysAssert', () => {
...
+    try {
+      new Assert().EqualKeys(['foo']).validate({});
+
+      should.fail();
+    } catch (e) {
+      e.show().violation.difference.should.eql(['foo']);
+    }
+  });
+
+  it('should allow setting no keys', () => {
```
Comment (nunofgs) [inline test/asserts/equal-keys-assert_test.js]: What are we setting here? WDYT of rewording to something along the lines of:

    'should allow `keys` to be undefined'

Comment (ruimarinho) [inline test/asserts/equal-keys-assert_test.js]: :+1:

**Comment by ruimarinho** [comment]
@nunofgs I think the README is correct, but since the outcome might be different for a certain input value, I think publishing as 2.0.0 is enough.

I agree with your comment regarding `new Assert().EqualKeys().validate({});` not throwing.

**Comment by nunofgs** [comment]
:+1:

**Comment by ricardogama** [comment]
LGTM

**Pull request merged**